common/src/main/java/org/geysermc/floodgate/FloodgatePlayer.java

```diff
@@ -37,10 +37,10 @@
      */
     private UUID javaUniqueId;
 
-    FloodgatePlayer(BedrockData data) {
+    FloodgatePlayer(BedrockData data, String prefix) {
         version = data.getVersion();
         username = data.getUsername();
-        javaUsername = "*" + data.getUsername().substring(0, Math.min(data.getUsername().length(), 15));
+        javaUsername = prefix + data.getUsername().substring(0, Math.min(data.getUsername().length(), 15));
```

Comment:
You should limit the prefix to one character or change the max allowed length of an username according to the prefix length